src/openlrr/game/interface/TextMessages.cpp

```diff
@@ -22,33 +32,301 @@ LegoRR::Text_Globs & LegoRR::textGlobs = *(LegoRR::Text_Globs*)0x00504190;
 #pragma region Functions
 
 // <LegoRR.exe @0046aab0>
-//void __cdecl LegoRR::Text_Load(Gods98::TextWindow* textOnlyWindow, Gods98::TextWindow* textImageWindow, sint32 imageX, sint32 imageY, real32 pauseTime);
+void __cdecl LegoRR::Text_Load(Gods98::TextWindow* textOnlyWindow, Gods98::TextWindow* textImageWindow, sint32 imageX, sint32 imageY, real32 pauseTime)
+{
+	// -FC, copy params to global object
+	textGlobs.textOnlyWindow = textOnlyWindow;
+	textGlobs.textImageWindow = textImageWindow;
+	textGlobs.TextImagePosition.x = imageX;
+	textGlobs.TextImagePosition.y = imageY;
+	textGlobs.textCount = Text_Type_Count;
+	textGlobs.currType = Text_Type_Count;
+	// -FC, now just copy over the strings
+	const char *TextLocs[] = {
+		"Text_Walk",
+		"Text_Drill",
+		"Text_NoDrill",
+		"Text_CantDrill",
+		"Text_Reinforce",
+		"Text_NoReinforce",
+		"Text_CantReinforce",
+		"Text_PickupCrystal",
+		"Text_Encyclopedia",
+		"Text_RadarHelp",
+		"Text_BuildHelp",
+		"Text_SelectionHelp",
+		"Text_Drive",
+		"Text_CantDrive",
+		"Text_MakeTeleporter",
+		"Text_CannotPlaceBuilding",
+		"Text_CrystalOreDisabled",
+		"Text_CrystalFound",
+		"Text_CavernDiscovered",
+		"Text_OreFound",
+		"Text_AirSupplyLow",
+		"Text_AirSupplyRunningOut",
+		"Text_SpaceToContinue",
+		"Text_GameCompleted",
+		"Text_ManTrained",
+		"Text_UnitUpgraded",
+	};
+	std::copy(std::begin(TextLocs), std::end(TextLocs), std::begin(textGlobs.textName));
```

Comment:
Although I appreciate the use of C++ here, LRR has a common way of defining enum/string names that's used in a lot of different modules. These ensure that the string is always stored at the index of the enum's value. I don't mind autogenerating these for anyone if they don't want to write it by hand, since writing by hand sounds awful.
```cpp
Text_RegisterName(Text_Walk);
Text_RegisterName(Text_Drill);
Text_RegisterName(Text_NoDrill);
Text_RegisterName(Text_CantDrill);
Text_RegisterName(Text_Reinforce);
Text_RegisterName(Text_NoReinforce);
Text_RegisterName(Text_CantReinforce);
Text_RegisterName(Text_PickupCrystal);
Text_RegisterName(Text_Encyclopedia);
Text_RegisterName(Text_RadarHelp);
Text_RegisterName(Text_BuildHelp);
Text_RegisterName(Text_SelectionHelp);
Text_RegisterName(Text_Drive);
Text_RegisterName(Text_CantDrive);
Text_RegisterName(Text_MakeTeleporter);
Text_RegisterName(Text_CannotPlaceBuilding);
Text_RegisterName(Text_CrystalOreDisabled);
Text_RegisterName(Text_CrystalFound);
Text_RegisterName(Text_CavernDiscovered);
Text_RegisterName(Text_OreFound);
Text_RegisterName(Text_AirSupplyLow);
Text_RegisterName(Text_AirSupplyRunningOut);
Text_RegisterName(Text_SpaceToContinue);
Text_RegisterName(Text_GameCompleted);
Text_RegisterName(Text_ManTrained);
Text_RegisterName(Text_UnitUpgraded);
```